lib/TH/generic/THTensorMath.c

```diff
@@ -1958,17 +1958,21 @@ LAB_IMPLEMENT_BASIC_FUNCTION(abs,abs)
 
 #if defined(TH_REAL_IS_BYTE)
 
-#define TENSOR_IMPLEMENT_LOGICAL_SUM(NAME, OP, INIT_VALUE) \
-  int THTensor_(NAME)(THTensor *tensor) \
-  { \
-    THArgCheck(tensor->nDimension > 0, 1, "empty Tensor"); \
-    int sum = INIT_VALUE;                               \
-    TH_TENSOR_APPLY(real, tensor, sum OP *tensor_data;); \
-    return sum; \
-  }
+int THTensor_(logicalall)(THTensor *tensor)
+{
+  THArgCheck(tensor->nDimension > 0, 1, "empty Tensor");
+  int sum = 1;
+  TH_TENSOR_APPLY(real, tensor, sum = sum && *tensor_data;);
+  return sum;
+}
 
-TENSOR_IMPLEMENT_LOGICAL_SUM(logicalall, &=, 1)
-TENSOR_IMPLEMENT_LOGICAL_SUM(logicalany, |=, 0)
```

Comment:
Is it not sufficient to change `&=` to `&&` and `|=` to `||` here? This would use the correct logical operators rather than bitwise operators. `sum` would probably need to be a bool in this case.